editor/js/commands/SetMaterialMapCommand.js

```diff
@@ -10,15 +10,16 @@
  * @constructor
  */
 
-var SetMaterialMapCommand = function ( object, mapName, newMap ) {
+var SetMaterialMapCommand = function ( object, mapName, newMap, materialSlot ) {
 
 	Command.call( this );
 	this.type = 'SetMaterialMapCommand';
 	this.name = 'Set Material.' + mapName;
 
 	this.object = object;
 	this.mapName = mapName;
-	this.oldMap = ( object !== undefined ) ? object.material[ mapName ] : undefined;
+	this.material = ( Array.isArray( object.material ) ) ? object.material[ materialSlot ] : object.material;
```

Comment:
Is `this.material` accessed from outside?

Maybe it would be better to follow the other command classes?
https://github.com/mrdoob/three.js/blob/dev/editor/js/commands/SetMaterialColorCommand.js

Reply:
I'll fix on my end.

Reply:
Actually, I think I prefer your approach, I'll change the other ones.

Reply:
Never mind 😇